service/controller/resource_set.go

```diff
@@ -488,7 +486,7 @@ func NewResourceSet(config ResourceSetConfig) (*controller.ResourceSet, error) {
 			return nil, microerror.Mask(err)
 		}
 
-		azureClients, err := client.NewAzureClientSetWithAuthorizer(authorizer, subscriptionID, partnerID)
+		azureClients, err := client.NewAzureClientSet(organizationAzureClientCredentialsConfig, subscriptionID, partnerID)
```

Comment:
With the new code we will create an Azure client set using:
- Organization/Customer credentials when there is clientid/clientsecret defined in organization secret (same as we do now)
- GiantSwarm multi tenant service principal (the one passed as parameter to operator) when no clientid/clientsecret are defined in organization secret